pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -796,11 +798,19 @@ func (c *Cluster) initMonIPs(mons []*monConfig) error {
 			}
 			m.PublicIP = node.Address
 		} else {
-			serviceIP, err := c.createService(m)
+			monService, err := c.createService(m)
 			if err != nil {
 				return errors.Wrap(err, "failed to create mon service")
 			}
-			m.PublicIP = serviceIP
+			if c.spec.Network.MultiClusterService.Enabled {
```

Comment:
We want existing mons to stay on their same IP even if the setting changes. I believe `m.PublicIP` would already be set from the `clusterInfoToMonConfigWithExclude()` method, so you might try this:
```suggestion
			if m.PublicIP == "" && c.spec.Network.MultiClusterService.Enabled {
```

Reply:
Tested `m.PublicIP == ""`. It works.  
The osds get updated first to use external IPs. After that, mon health check fails the mons one by one to use external IP.